refactor/user-type: User 데코레이터 수정 및 컨트롤러 타입 안정성 강화

Na1gae opened the pull request:
* Auth/Tree Controller를 중심으로 타입 지정 추가

* 가드 변경에 따른 테스트 코드 수정

* 기존 코드에서 any 타입으로 처리되던 사용자 객체에 명확한 타입 지정
=> 데코레이터를 사용하여 특정 속성을 주입하도록

* UseCase 테스트 데이터의 socialProvider값이 ENUM타입과 일치하지 않는 부분 수정

Na1gae committed:
feat: User 데코레이터에 특정 속성 추출 기능 추가

Na1gae committed:
refactor: User 데코레이터 변수 수정

Na1gae committed:
test: 인증 방식 변경에 따른 컨트롤러 테스트코드 업데이트

Na1gae committed:
chore: 테스트 데이터의 소셜 프로바이더값 수정

Na1gae committed:
test: 컨트롤러 테스트 코드 수정

Auth와 Tree 컨트롤러의 테스트에서 발생하던 오류를 수정했습니다.
* DTO 유효성을 통과하도록 형식 조절
* 반환 Mock 데이터를 실제 값과 동일하게 변경

Na1gae committed:
refactor: Auth/User Controller 타입 명시

각 컨트롤러의 데코레이터 파라미터에 명확한 타입을 지정하고, 필요한 값만 직접 주입하도록 리팩토링했습니다.

juwoong reviewed (changes requested):
이거 어짜피 User 객체가 존재하는지 db를 통해 validation 해야 하는데, 그럴거면 그냥 userId 쓰지말고 User Entity를 정의하고 그걸 request안에 넣고 쓰도록 하는게 좋아보입니다.

Na1gae committed:
chore:  UserParam에서 빠진 데이터 추가

Na1gae committed:
chore: Prisma의 UserType에서 UserParam으로 타입 변경

Na1gae committed:
refactor: UserParam을 활용한 로직 최적화

AccessTokenGuard를 이용해 받아온 UserParam을 서비스계층까지 전달하도록 수정했습니다.
-> 일부 불필요한 DB 조회가 제거되었습니다

Na1gae committed:
chore: UserParam의 firebaseUid를 불러오도록 수정

Na1gae committed:
refactor: UserParam에 firebaseUid가 포함되도록 수정

Na1gae committed:
test: UserParam 형식에 맞게 mockUser 수정

juwoong committed:
Merge branch 'develop' into refactor/user-type
